### ISSUE TITLE
Boost mode with multiple y-axes ignores yAxis min/max values, allowing axes' series to overlap unexpectedly

### ISSUE THREAD

**Issue by sjdemartini** [highcharts/highcharts#19039]
#### Expected behaviour

When using multiple `yAxis` option objects to have separate "stacked" y-axes, the `yAxis` `min` and `max` values should prevent any all series from overflowing beyond those min/max values (clipping them visually), regardless of whether the series are using boost mode or not.

#### Actual behaviour

If a series is using boost mode, the `yAxis` `min` and `max` values are not respected when the series is assigned to that axis. The series will extend into any/all adjacent yAxes, unexpectedly going beyond the boundaries of the intended axis.

#### Live demo with steps to reproduce

https://jsfiddle.net/vyq284os/4/

In this example, there are 4 `yAxis` configurations provided, all of which have a `min` set at 0 and a `max` set at 3. Each axis has a single sine wave series added to it which varies between -10 and 10. As can be seen in the screenshot, when `boost` is `enabled: true`, any `yAxis` that is "on top of" any below will allow its boosted series line to extend beyond the `min` specified for that axis and cross into the axes below it. And any `yAxis` that is below any others will allow its boosted series to extend beyond the `max` specified for that axis and cross into the axes above it.

Notice that when boost mode is disabled (overall or for a specific series, as with the `non-boost`-labeled series here), the `yAxis` `min` and `max` values _are_ respected, so the series do **not** extend beyond the axis extremes (as expected). Note that zooming in the y-axis direction has the same behavior as setting `min`/`max` in this way, where boosted series will go beyond the bounds of the zoomed y-axis that they're assigned to.

![image](https://github.com/highcharts/highcharts/assets/1647130/881bcfc5-c71e-46ac-bc3d-390ee247909e)

#### Product version

Highcharts latest version 11.0.1. (Also present in Highcharts 10. Unknown if present in earlier versions.)

#### Affected browser(s)

Confirmed bug in both Chrome and Firefox

**Comment by bm64** [comment]
Hello @sjdemartini and thanks for reporting the issue. 

The issue is most likely related to https://github.com/highcharts/highcharts/issues/11906

**Comment by TorsteinHonsi** [comment]
Closed by https://github.com/highcharts/highcharts/commit/895a814becc081031548f84200245d8d3935c19d. Sorry for all the noise in that commit, it was a merge gone bad and redone.

**Comment by TorsteinHonsi** [comment]
Demo with the fix applied: https://jsfiddle.net/highcharts/596ghw70/

**Comment by sjdemartini** [comment]
@TorsteinHonsi Thanks for fixing this! Excited to use an updated version that has the fix. For what it's worth, the demo Fiddle does not load for me (seems the JS isn't loading with the paths provided). Also, I see the fix was since commented out. Should this issue be re-opened until it's fixed again with testing later?

**Comment by TorsteinHonsi** [comment]
Yes, sorry about that. I messed up my local branch, decided to push it anyway, regretted, commented it out, and decided to do it properly 😅.

Now I have created a proper PR at https://github.com/highcharts/highcharts/pull/19724. It will allow the CI to catch potential visual regressions against the master.

I'm not able to get the preview to work right now, but here's how it looks:
<img width="768" alt="Skjermbilde 2023-09-08 kl  22 35 45" src="https://github.com/highcharts/highcharts/assets/227836/7ffc83aa-a851-402d-852c-0ca52a9e2d85">
